packages/playwright-core/src/server/browser.ts

```diff
@@ -82,7 +83,14 @@ export abstract class Browser extends SdkObject {
 
   async newContext(metadata: CallMetadata, options: channels.BrowserNewContextParams): Promise<BrowserContext> {
     validateBrowserContextOptions(options, this.options);
+    let clientCertificatesProxy: ClientCertificatesProxy | undefined;
+    if (options.clientCertificates?.length) {
```

Comment:
Let's extract lines 87-91 as a helper method to reuse for persistent context, put it right next to `validateBrowserContextOptions` and always call them together.